sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -597,6 +598,12 @@ Scheduler::GraphBuilder::findAllocaForReq(MemObjRecord *Record,
 }
 
 static bool checkHostUnifiedMemory(const ContextImplPtr &Ctx) {
+  if (const char *HUMConfig = SYCLConfig<SYCL_HOST_UNIFIED_MEMORY>::get()) {
+    if (std::strcmp(HUMConfig, "0") == 0)
+      return Ctx->is_host();
+    if (std::strcmp(HUMConfig, "1") == 0)
+      return true;
```

Comment:
I'd prefer `strncmp` here

Reply:
I would prefer to keep it as `strcmp` to have only 0/1 as valid values and prevent something like `SYCL_HOST_UNIFIED_MEMORY=01 ./a.out` from having an effect.

Reply:
In the config you explicitly specified 1 byte to be used for storage. So, in that case only the first character should be read.

Reply:
The 1 byte storage is only used to store a configuration value if it is set via a config file, it's completely bypassed if an environment variable is used instead. So in the example I gave above, the config value is not truncated.